### PR TITLE
chore: unify the event time of messages to drop

### PR DESCRIPTION
**Pull request by KeranYang**
Set the event time of messages to drop to epoch(0) - 1ms across all our SDKs.

### CODE REVIEW

File: pynumaflow/sourcetransformer/_dtypes.py [hunk 1]
```diff
@@ -12,6 +12,15 @@
 
 @dataclass(init=False)
 class Message:
+    """
+    event_time_to_drop 1969-12-31 00:00:00 +0000 UTC is set to be slightly
+    earlier than Unix epoch -1 (1969-12-31 23:59:59.999 +0000 UTC)
+    As -1 is used on Numaflow to indicate watermark is not available, event_time_to_drop is
+    used to indicate that the message is dropped hence, excluded from watermark calculation
+    """
```
Comment (ab93) [inline pynumaflow/sourcetransformer/_dtypes.py:20]: Let's add this after the class docstring, in the same space

File: pynumaflow/sourcetransformer/_dtypes.py [hunk 2]
```diff
@@ -12,6 +12,15 @@
...
+
+    event_time_to_drop = datetime(1969, 12, 31, 0, 0, 0, tzinfo=timezone.utc)
```
Comment (ab93) [inline pynumaflow/sourcetransformer/_dtypes.py:22]: Do we really need this as a new field? Can't we just store this as a constant, and assign?

File: pynumaflow/sourcetransformer/_dtypes.py [hunk 3]
```diff
@@ -44,7 +53,7 @@ def __init__(
 
     @classmethod
     def to_drop(cls: type[M]) -> M:
-        return cls(b"", datetime(1, 1, 1, 0, 0), None, [DROP])
+        return cls(b"", Message.event_time_to_drop, None, [DROP])
```
Comment (ab93) [inline pynumaflow/sourcetransformer/_dtypes.py:56]: Something like:
```suggestion
        return cls(b"", EVENT_TIME_TO_DROP, None, [DROP])
```

**Review by vigith — COMMENTED** [review]
let's make it a `_const`

**Review by vigith — APPROVED** [review]

**Review by ab93 — APPROVED** [review]

**Pull request merged**